Dockerfile

```diff
@@ -2,12 +2,8 @@ FROM balena/open-balena-base:v4.4.2
 
 EXPOSE 80
 
-COPY package.json package-lock.json /usr/src/app/
-RUN npm ci --unsafe-perm --production && npm cache clean --force
-
 COPY . /usr/src/app
-
-RUN npx tsc --noEmit
+RUN npm ci --unsafe-perm --production && npm cache clean --force
```

Comment:
Why did you remove the separate
```
COPY package.json package-lock.json /usr/src/app/
RUN npm ci --unsafe-perm --production && npm cache clean --force
```
?
The new version reduces the caching a lot so will slow down builds

Reply:
`npm` will invoke the `prepare` script during `npm ci` which triggers a build of the project. Without first copying the whole project into `/usr/src/app` there are no files to compile (or a tsconfig.json) and it fails. I don't like this TBH, because we're compiling sources during build but aren't really using them -- open-balena-api still starts up like resin-api, compiling on-demand. Any ideas?

Reply:
There's a `--ignore-scripts` for `npm i` that might work for ci?

Reply:
Ah, interesting, I'll take a look.

Reply:
That worked nicely 👍